deepchem/dock/pose_generation.py

```diff
@@ -79,6 +79,166 @@ def generate_poses(self,
     raise NotImplementedError
 
 
+class GninaPoseGenerator(PoseGenerator):
+  """Use GNINA to generate binding poses.
+
+  This class uses GNINA (a deep learning framework for molecular
+  docking) to generate binding poses. It downloads the GNINA
+  executable to DEEPCHEM_DATA_DIR (an environment variable you set)
+  and invokes the executable to perform pose generation.
+
+  GNINA uses pre-trained convolutional neural network (CNN) scoring
+  functions to rank binding poses based on learned representations of
+  3D protein-ligand interactions. It has been shown to outperform
+  AutoDock Vina in virtual screening applications [1]_.
+
+  If you use the GNINA molecular docking engine, please cite the relevant
+  papers: https://github.com/gnina/gnina#citation
+  The primary citation for GNINA is [1]_.
+
+  References
+  ----------
+  .. [1] M Ragoza, J Hochuli, E Idrobo, J Sunseri, DR Koes.
+  "Protein–Ligand Scoring with Convolutional Neural Networks."
+  Journal of chemical information and modeling (2017).
+
+  Notes
+  -----
+  * GNINA requires CUDA >= 10.1 for fast CNN scoring.
+  * Almost all dependencies are included in the most compatible way
+    possible, which reduces performance. Build GNINA from source
+    for production use.
+
+  """
+
+  def __init__(self):
+    """Initialize GNINA pose generator."""
+
+    data_dir = get_data_dir()
+    if platform.system() == 'Linux':
+      url = "https://github.com/gnina/gnina/releases/download/v1.0/gnina"
+      filename = 'gnina'
+      self.gnina_dir = data_dir
+      self.gnina_cmd = os.path.join(self.gnina_dir, filename)
+    else:
+      raise ValueError(
+          "Unknown operating system. Try using a cloud platform to run this code instead."
```

Comment:
Perhaps "Gnina currently only runs on Linux. Try using a cloud platform to run this code instead."